curriculum/challenges/english/blocks/lab-game-character-stats/68cabc534d2e741f33a3fe1d.md

```diff
@@ -0,0 +1,536 @@
+})
+```
+
+You should return `self._level` in the `level` method.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").find_function("level").has_return("self._level")`)
+})
+```
+
+The `level` method should have a `@property` decorator.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").find_function("level").has_decorators("property")`) 
+})
+```
+
+You should have a `level_up` method in the `GameCharacter` class.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").has_function("level_up")`) 
+})
+```
+
+The `level_up` method should have only one parameter, `self`.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").find_function("level_up").has_args("self")`) 
+})
+```
+
+You should increase `self._level` by `1` in the `level_up` method.
+
+```js
+({ 
+  test: () => runPython(`
+    node = _Node(_code).find_class("GameCharacter").find_function("level_up")
+    assert (
+        node.has_stmt("self._level += 1") or
+        node.has_stmt("self._level = self._level + 1")
+    )
+`)
+})
+```
+
+You should set `self.health` to `100` in the `level_up` method.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").find_function("level_up").has_stmt("self.health = 100")`) 
+})
+```
+
+You should set `self.mana` to `50` in the `level_up` method.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").find_function("level_up").has_stmt("self.mana = 50")`) 
+})
+```
+
+You should use an f-string to print `f"{self._name} leveled up to {self._level}"`.
+
+```js
+({
+  test: () => runPython(`
+    func = _Node(_code).find_class("GameCharacter").find_function("level_up")
+    assert (
+      func.has_call('print(f"{self._name} leveled up to {self._level}")') or
+      func.has_call("print(f'{self._name} leveled up to {self._level}')")
+    )
+  `)
+})
+```
+
+You should have a `__str__` method in the `GameCharacter` class.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").has_function("__str__")`) 
+})
+```
+
+The `__str__` method should have only one parameter, `self`.
+
+```js
+({ 
+    test: () => runPython(`assert _Node(_code).find_class("GameCharacter").find_function("__str__").has_args("self")`) 
+})
+```
+
+You should return the character's stats with an f-string: `f"Name: {self.__name}\nLevel: {self.__level}\nHealth: {self.__health}\nMana: {self.__mana}"`.
+
+```js
+({ 
+  test: () => runPython(`
+    node = _Node(_code).find_class("GameCharacter").find_function("__str__")
+    assert (
+        node.has_return("f'Name: {self._name}\\\\nLevel: {self._level}\\\\nHealth: {self._health}\\\\nMana: {self._mana}'") or
+        node.has_return('f"Name: {self._name}\\\\nLevel: {self._level}\\\\nHealth: {self._health}\\\\nMana: {self._mana}"')
+    )
```

Comment:
```suggestion
Your `__str__` method should return a string with the character's stats using the provided format.

```js
({ 
  test: () => runPython(`
import io
import sys
        
captured_output = io.StringIO()
sys.stdout = captured_output
        
gc = GameCharacter("test")
gc.mana = 15
gc.health = 77
print(gc)
        
sys.stdout = sys.__stdout__
output = captured_output.getvalue()

expected = """Name: test
Level: 1
Health: 77
Mana: 15"""
assert expected in output
```